keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -0,0 +1,497 @@
+            Defaults to None.
+        save_metrics_state: Boolean, whether to include stateful metrics
+            variables in the checkpoint. Defaults to False.
+        post_finalization_callback: Callable, function to call after async
+            checkpointing operations complete. Defaults to None.
+        save_decision_policy: orbax.checkpoint.SaveDecisionPolicy object to
+            control when checkpoints are saved. If None, defaults to saving
+            every epoch for save_freq="epoch" or every save_freq batches.
+            Defaults to None.
+    """
+
+    def __init__(
+        self,
+        directory,
+        monitor="val_loss",
+        verbose=0,
+        save_best_only=False,
+        mode="auto",
+        save_freq="epoch",
+        max_to_keep=5,
+        keep_period=None,
+        initial_value_threshold=None,
+        save_optimizer_state=True,
+        save_on_background=True,
+        save_metadata=None,
+        save_data_iterator=None,
+        save_metrics_state=False,
+        post_finalization_callback=None,
+        save_decision_policy=None,
+    ):
+        # Ensure orbax is available
+        ocp.initialize()
+
+        # Initialize MonitorCallback for handling 'monitor', 'mode', 'best'
+        # logic
+        super().__init__(monitor, mode, initial_value_threshold)
+
+        self.directory = directory
+        self.verbose = verbose
+        self.save_best_only = save_best_only
+        self.save_freq = save_freq
+        self.save_optimizer_state = save_optimizer_state
+        self.save_on_background = save_on_background
+        self.save_metadata = save_metadata
+        self.save_data_iterator = save_data_iterator
+        self.save_metrics_state = save_metrics_state
+        self.post_finalization_callback = post_finalization_callback
+        self.save_decision_policy = save_decision_policy
+        self._batches_seen_since_last_saving = 0
+        self._last_batch_seen = 0
+        self._current_epoch = 0  # Keep track of epoch
+        self._total_batches_seen = 0  # Global batch counter for step tracking
+
+        if self.save_freq != "epoch" and not isinstance(self.save_freq, int):
+            raise ValueError("Unrecognized save_freq")
+
+        # Set up save_decision_policy if not provided
+        if save_decision_policy is None:
+            # Let Keras handle all save decisions - configure Checkpointer
+            # to save unconditionally when save_pytree/save_pytree_async
+            # is called
+            class _AlwaysSavePolicy(
+                ocp.training.save_decision_policies.SaveDecisionPolicy
+            ):
+                def should_save(
+                    self, current_step_info, previous_steps=None, context=None
+                ):
+                    return True
+
+            save_decision_policy = _AlwaysSavePolicy()
+
+        # --- Orbax Checkpointer Setup (V1 API) ---
+        # Map V0 options to V1 parameters
```

Comment:
`# Map V0 options to V1 parameters`

I'm not following this comment. We're not using anything V0. I guess just remove the comment.

Reply:
Done